mlflow/entities/trace_info.py

```diff
@@ -71,3 +74,35 @@ def from_proto(cls, proto):
             request_metadata={attr.key: attr.value for attr in proto.request_metadata},
             tags={tag.key: tag.value for tag in proto.tags},
         )
+
+    @staticmethod
+    def validate_tag_key_value(key: Any, value: Any):
+        """
+        Validate tag key and value.
+        """
+        if not key or not isinstance(key, str):
+            raise MlflowException(
+                f"A key for a trace tag must be a non-empty string. Got: {key}",
+                INVALID_PARAMETER_VALUE,
+            )
+
+        if not isinstance(value, str):
+            raise MlflowException(
+                f"A value for a trace tag must be a string. Got: {value} for the key '{key}'",
+                INVALID_PARAMETER_VALUE,
+            )
+
+        if len(key) > MAX_CHARS_IN_TRACE_INFO_METADATA_AND_TAGS:
+            raise MlflowException(
+                f"A key for a trace tag exceeds the maximum allowed length of "
+                f"{MAX_CHARS_IN_TRACE_INFO_METADATA_AND_TAGS} characters. Got: {len(key)}",
+                INVALID_PARAMETER_VALUE,
+            )
+
+        if len(value) > MAX_CHARS_IN_TRACE_INFO_METADATA_AND_TAGS:
+            raise MlflowException(
+                f"A value for a trace tag exceeds the maximum allowed length of "
+                f"{MAX_CHARS_IN_TRACE_INFO_METADATA_AND_TAGS} characters. "
+                "Got: {len(value)} for the key '{key}'",
+                INVALID_PARAMETER_VALUE,
+            )
```

Comment:
Do we need this validation and the checks if we're automatically truncating these?